ConsoleHabitTracker.kraven88/Menu.cs

```diff
@@ -0,0 +1,296 @@
+        }
+    }
+    
+    internal bool HabitMenu(Habit habit)
+    {
+        var isRunning = true;
+        while (isRunning) 
+        {
+            DisplayHabitMenuItems(habit);
+            var select = Console.ReadKey(true).KeyChar;
+            isRunning = select switch
+            {
+                '1' => UpdateDailyProgress(habit),
+                '2' => ViewCurrentProgress(habit),
+                '3' => ViewEntireProgress(habit),
+                '4' => SetDailyGoal(habit),
+                '5' => DeleteCurrentProgress(habit),
+                '6' => DeleteAllProgress(habit),
+                '0' => false,
+                _ => true,
+            };
+        }
+        return true;
+    }
+    
+    private void HeaderText(string habitName = "Habit")
+    {
+        Console.Clear();
+        Console.WriteLine("============================");
+        Console.WriteLine($"   {habitName} Logger by kraven   ");
+        Console.WriteLine("============================" + nl);
+    }
+    
+    private bool SelectHabit(List<Habit> habits)
+    {
+        Console.Write(nl + "Select: ");
+        var select = Console.ReadLine()!.Trim().ToLower();
+
+        if (select == "new") 
+            return CreateNewHabit();
+
+        else if (select == "0") 
+            return false;
+
+        else if (int.TryParse(select, out int number) && number > 0 && number <= habits.Count)
+        {
+            var habit = habits.Where(x => x.Id == number).First();
+            return HabitMenu(habit);
+        }
+        else 
+            return true;
+
+    }
+    
+    private bool SetDailyGoal(Habit habit)
+    {
+        HeaderText(habit.Name);
+        
+        Console.WriteLine($"{habit.Name} current daily goal is {habit.CurrectGoal} {habit.UnitOfMeasurement}");
+
+        var newGoal = AskForInteger("Enter new daily goal (0 to leave unchanged): ");
+
+        if (newGoal == 0) return true;
+        else
+        {
+            db.SaveNewGoal(habit, newGoal);
+            habit.CurrectGoal = newGoal;
+        }
+
+        return true;
+        
+    }
+    
+    private bool UpdateDailyProgress(Habit habit)
+    {
+        HeaderText(habit.Name);
+        var today = DateOnly.FromDateTime(DateTime.Now).ToString("dd.MM.yyyy");
+
+        var newProgress = AskForInteger($"Please enter the number of ({habit.UnitOfMeasurement}) since your last update: ");
+        if (newProgress == 0) return true;
+        else
+        {
+            db.SaveProgress(habit, today, newProgress);
+            Console.WriteLine("Progress succesfuly updated.");
+            Console.ReadKey();
+            return true;
+        }
+    }
+    
+    private bool ViewCurrentProgress(Habit habit)
+    {
+        HeaderText(habit.Name);
+        var today = DateOnly.FromDateTime(DateTime.Now).ToString("dd.MM.yyyy");
+
+        //db.SaveProgress(habit, today, 0);   // This ensures that, when selected, Current Progress will always include current day data
```

Comment:
Remove unused code. Add //TODO if you want to remind yourself of a future task.

Reply:
Just remove. 
I believe I commented this out, because it was unnecessary, and I forgot to delete the comment entirely at the end.